lib/framework/data_context.dart

```diff
@@ -546,45 +546,52 @@ class EnsembleStorage with Invokable {
   }
 }
 
-/// singleton handling utils
 class EnsembleUtils with Invokable {
-  static final EnsembleUtils _obj = EnsembleUtils._();
-  EnsembleUtils._();
-  factory EnsembleUtils() => _obj;
-
   @override
-  Map<String, Function> getters() => {'getCountries': () => allCountries};
+  Map<String, Function> getters() => {};
 
   @override
   Map<String, Function> methods() => {
-        'getCountries': (value) {
+        'getCountries': () => allCountries,
+        'getCountry': (value) {
           String val = Utils.getString(value, fallback: "");
-          return CountryManager().getCountryData(val);
+          return getCountry(val);
         },
+        'findCountry': (value) {
+          String val = Utils.getString(value, fallback: "");
+          return findCountry(val);
+        }
       };
 
   @override
   Map<String, Function> setters() => {};
-}
 
-class CountryManager {
-  static final CountryManager _instance = CountryManager._();
-  CountryManager._();
-  factory CountryManager() => _instance;
+  List<Map<String, dynamic>> getCountry(String val) {
+    String input = val.toLowerCase().trim();
+    List<Map<String, dynamic>> alpha2List = [];
```

Comment:
We shouldn't have to iterate through the list if the # of characters is not 2 - it's wasteful.

Even better, let's search by alpha-2 if 2 characters, and alpha-3 if 3 characters.